pkg/hasher/hashstructure/hashstructure.go

```diff
@@ -0,0 +1,326 @@
+/**
+This is a slightly modified version of github.com/mitchellh/hashstructure
+
+The reduced feature set is to avoid using reflect.Value.Interface() that can cause data races when
+used with google.golang.org/protobuf
```

Comment:
this isn't a copy of latest master.. can we include the commit hash this was copied from?

also can we add some more color / description to how this can cause races with protobuf? (specifically `Clone()` and `Marshal()`)